### PR TITLE
feat: add use strict for iife

### PR DESCRIPTION
**Pull request by underfin**

### CODE REVIEW

**Comment by 7086cmd** [comment]
I just added it in #1704 😂.

File: packages/rolldown/tests/fixtures/output/format/iife/exports/_config.ts [hunk 1]
```diff
@@ -13,6 +13,8 @@ export default defineTest({
     expect(output.output[0].code).toMatchInlineSnapshot(`
       "(function(exports, node_path) {
```
Comment (hyf0) [inline packages/rolldown/tests/fixtures/output/format/iife/exports/_config.ts:15]: Could you also fix this empty line in another PR?

File: crates/rolldown/src/ecmascript/format/cjs.rs [hunk 2]
```diff
@@ -22,17 +22,7 @@ pub fn render_cjs(
   let mut concat_source = ConcatSource::default();
 
   // Add `use strict` directive if needed. This must come before the banner, because users might use banner to add hashbang.
-  let are_modules_all_strict = ctx
```
Comment (underfin) [inline crates/rolldown/src/ecmascript/format/cjs.rs:25]: cc @hyf0 The rollup `banner` should be prepend at `use strict` for cjs, you could see [amd example]( https://rollupjs.org/repl/?version=4.19.0&shareable=JTdCJTIyZXhhbXBsZSUyMiUzQSUyMjAxJTIyJTJDJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMmNvZGUlMjIlM0ElMjIlMkYlMkYlMjBOQU1FRCUyMEVYUE9SVFMlNUNuJTJGJTJGJTIwVGhlcmUlMjBhcmUlMjBtYW55JTIwd2F5cyUyMHRvJTIwZXhwb3J0JTIwYmluZGluZ3MlNUNuJTJGJTJGJTIwZnJvbSUyMGFuJTIwRVMyMDE1JTIwbW9kdWxlJTVDbmV4cG9ydCUyMHZhciUyMGZvbyUyMCUzRCUyMDElM0IlNUNuJTVDbmV4cG9ydCUyMGZ1bmN0aW9uJTIwYmFyKCklMjAlN0IlNUNuJTVDdCUyRiUyRiUyMHRyeSUyMGNoYW5naW5nJTIwdGhpcyUyMHRvJTIwJTYwZm9vJTJCJTJCJTYwJTVDbiU1Q3QlMkYlMkYlMjB3aGVuJTIwZ2VuZXJhdGluZyUyMENvbW1vbkpTJTVDbiU1Q3RyZXR1cm4lMjBmb28lM0IlNUNuJTdEJTVDbiU1Q25mdW5jdGlvbiUyMGJheigpJTIwJTdCJTVDbiU1Q3RyZXR1cm4lMjBiYXIoKSUzQiU1Q24lN0QlNUNuJTVDbmV4cG9ydCUyMCU3QiUyMGJheiUyMCU3RCUzQiU1Q25leHBvcnQlMjAqJTIwZnJvbSUyMCcuJTJGcXV4JyUzQiUyMiUyQyUyMmlzRW50cnklMjIlM0F0cnVlJTJDJTIybmFtZSUyMiUzQSUyMm1haW4uanMlMjIlN0QlMkMlN0IlMjJjb2RlJTIyJTNBJTIyZXhwb3J0JTIwY29uc3QlMjBxdXglMjAlM0QlMjAnUVVYJyUzQiUyMiUyQyUyMmlzRW50cnklMjIlM0FmYWxzZSUyQyUyMm5hbWUlMjIlM0ElMjJxdXguanMlMjIlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyb3V0cHV0JTIyJTNBJTdCJTIyZXNNb2R1bGUlMjIlM0ElMjJpZi1kZWZhdWx0LXByb3AlMjIlMkMlMjJleHBvcnRzJTIyJTNBJTIyYXV0byUyMiUyQyUyMmZvcm1hdCUyMiUzQSUyMmFtZCUyMiU3RCU3RCU3RA==)

**Review by hyf0 — APPROVED** [review]

File: crates/rolldown/src/ecmascript/format/cjs.rs (hunk 2 above)
Comment (7086cmd) [reply crates/rolldown/src/ecmascript/format/cjs.rs:25]: I will modify #1704 when the PR is merged, and fix the banner in it.

File: packages/rolldown/tests/fixtures/output/format/iife/exports/_config.ts (hunk 1 above)
Comment (underfin) [reply packages/rolldown/tests/fixtures/output/format/iife/exports/_config.ts:15]: Fixed.

File: crates/rolldown/src/ecmascript/format/cjs.rs (hunk 2 above)
Comment (hyf0) [reply crates/rolldown/src/ecmascript/format/cjs.rs:25]: @7086cmd Could you open another PR for fixing this banner order problem?

**Review by hyf0 — APPROVED** [review]

File: crates/rolldown/src/ecmascript/format/cjs.rs (hunk 2 above)
Comment (hyf0) [reply crates/rolldown/src/ecmascript/format/cjs.rs:25]: https://github.com/rolldown/rolldown/issues/1706

Comment (7086cmd) [reply crates/rolldown/src/ecmascript/format/cjs.rs:25]: See #1707.

**Pull request merged**